pytest/test_io_meshes.py

```diff
@@ -26,4 +36,113 @@ def test_STL(tmp_path):
     except ImportError:
         warn("VTK is not installed and thus has not been tested.")
 
+def test_from_meshio_pygmsh_WaveBot():
+    try:
+        T1=0.16
+        T2=0.37
+        r1=0.88
+        r2=0.35
+        with pygmsh.occ.Geometry() as geom:
+                cyl = geom.add_cylinder([0,0,0], [0,0,-T1], r1)
+                cone = geom.add_cone([0,0,-T1], [0,0,-T2], r1, r2)
+                geom.translate(cyl, [0, 0, offset])
+                geom.translate(cone, [0, 0, offset])
+                geom.boolean_union([cyl, cone])       
+                mesh = geom.generate_mesh(dim=2)
+        fb = FloatingBody.from_meshio(mesh)
+
+        vol = fb.mesh.volume
+        vol_exp = np.pi*T1*r1**2 + 1/3 * np.pi * T2 * (r2**2 + r2 * r1 + r1**2)
+
+        assert pytest.approx(vol_exp, rel=1e-1) == vol
+
+        fb.add_translation_dof((0,0,1))
+
+        test_matrix = xr.Dataset(coords={
+            'rho': rho_water,
+            'water_depth': [np.infty],          
+            'omega': omega,
+            'wave_direction': 0,
+            'radiating_dof': list(fb.dofs.keys()),
+            })
+
+        solver = BEMSolver()
+        data = solver.fill_dataset(test_matrix, 
+                                bodies=[fb],
+                                hydrostatics=True,
+                                mesh=True,
+                                wavelength=True,
+                                wavenumber=True)
+    except ImportError:
+        warn('pygmsh and/or meshio is not installed and thus from_meshio has not been tested')
```

Comment:
Does this work even when `import pygmsh` is outside of the `try` block?

Reply:
Haha good point. I didn't test it in an environment without `pygmsh`... with 021347600c5d16994b734e453fa588f28ad04e8b I have found a better way to do this, which results in `pytest` skipping tests if the optional dependency is not installed


```
$ pytest test_io_meshes.py 
================================================================ test session starts =================================================================
platform darwin -- Python 3.9.2, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /Users/rcoe/capytaine
plugins: hypothesis-6.9.1
collected 4 items                                                                                                                                    

test_io_meshes.py .sss                                                                                                                         [100%]

================================================================== warnings summary ==================================================================
test_io_meshes.py:45
  /Users/rcoe/capytaine/pytest/test_io_meshes.py:45: PytestUnknownMarkWarning: Unknown pytest.mark.pygmsh_functions - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.pygmsh_functions

test_io_meshes.py:86
  /Users/rcoe/capytaine/pytest/test_io_meshes.py:86: PytestUnknownMarkWarning: Unknown pytest.mark.pygmsh_functions - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.pygmsh_functions

test_io_meshes.py:121
  /Users/rcoe/capytaine/pytest/test_io_meshes.py:121: PytestUnknownMarkWarning: Unknown pytest.mark.pygmsh_functions - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.pygmsh_functions

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================================================== 1 passed, 3 skipped, 3 warnings in 3.55s ======================================================
```

Reply:
Nice. I did not know this feature from pytest. We should use that for the other optional dependencies, like quadpy.